automap/src/comm_layer/pcp_pmp_common/linux_specific.rs

```diff
@@ -120,55 +103,67 @@ Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
         )
     }
 
+    #[test]
+    fn find_routers_works_when_there_is_no_router_to_find() {
+        let route_n_output = "\
+        10.1.0.0/16 dev eth0 proto kernel scope link src 10.1.0.84 metric 100\n\
+        0.1.0.1 dev eth0 proto dhcp scope link src 10.1.0.84 metric 100\n\
+        168.63.129.16 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100\n\
+        169.254.169.254 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100\n\
+        172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown";
+        let find_routers_command = FindRoutersCommandMock::new(Ok(&route_n_output));
+
+        let result = linux_find_routers(&find_routers_command).unwrap();
+
+        assert_eq!(result.is_empty(), true)
+    }
+
     #[test]
     fn find_routers_command_produces_output_that_looks_right() {
         let subject = LinuxFindRoutersCommand::new();
 
         let result = subject.execute().unwrap();
 
         let lines = result.split('\n').collect::<Vec<&str>>();
-        assert_eq!("Kernel IP routing table", lines[0]);
-        let headings = lines[1]
-            .split(' ')
-            .filter(|s| s.len() > 0)
-            .collect::<Vec<&str>>();
-        assert_eq!(
-            headings,
-            vec![
-                "Destination",
-                "Gateway",
-                "Genmask",
-                "Flags",
-                "Metric",
-                "Ref",
-                "Use",
-                "Iface",
-            ]
+        assert!(
+            lines.len() > 1,
+            "Did not find more than two lines in this vector {:?}",
+            lines
         );
-        for line in &lines[3..] {
-            if line.len() == 0 {
-                continue;
-            }
-            let columns = line
-                .split(' ')
-                .filter(|s| s.len() > 0)
-                .collect::<Vec<&str>>();
-            for idx in 0..3 {
-                if IpAddr::from_str(columns[idx]).is_err() {
-                    panic!(
-                        "Column {} should have been an IP address but wasn't: {}",
-                        idx, columns[idx]
-                    )
-                }
-            }
-            for idx in 4..7 {
-                if columns[idx].parse::<u64>().is_err() {
-                    panic!(
-                        "Column {} should have been numeric but wasn't: {}",
-                        idx, columns[idx]
-                    )
-                }
-            }
-        }
+        let reg = ip_route_regex();
+        lines.iter().for_each(|line| {
+            assert!(reg.is_match(line), "Lines: {:?} line: {}", lines, line);
+        });
+    }
+
+    fn ip_route_regex() -> Regex {
+        let reg_for_ip = r"((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}";
+        Regex::new(&format!(
+            r#"{}|^{}(/\d+)?\s(dev|via)(\s\.+)\{{3,\}}"#,
```

Comment:
This regular expression probably isn't right. Review it.

This one is probably more correct: `r#"^{}(/\d+)?\s(dev|via)(\s.+){{3,}}"`